core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseFragmentActivityExtensions.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.base
+
+import android.content.Intent
+import android.view.ActionMode
+import android.view.Menu
+import androidx.appcompat.app.AppCompatActivity
+
+interface BaseFragmentActivityExtensions {
+  enum class Super {
+    ShouldCall,
+    DontCall
+  }
+
+  fun onActionModeStarted(actionMode: ActionMode, activity: AppCompatActivity): Super
```

Comment:
Kotlin Interfaces can have default implementations in the interface so perhaps the default should be simply returning `ShouldCall` and we should only override when we absolutely have to do something? I am not sure I fully get the meaning of these enums.

Typically in android for a chain of listeners that can short circuit you would pass `true` for a consumed event and `false` to allow the event to handled lower down the chain. This would come up if you had overlapping views with touchlisteners

Reply:
It was mainly for the onBackPressed method I think. Most of the time it should not call super but sometimes it should. It does not naturally return a boolean which means we have to handle the conditional super call on our own. 
```
override fun onBackPressed() {
    supportFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
      if (it.onBackPressed(this) == ShouldCall) {
        super.onBackPressed()
      }
    }
  }
```

Reply:
Maybe returning a boolean is better/more typical to android? I found ShouldCall/ShouldNotCall more readable.